Entities/Enemy.gd

```diff
@@ -251,10 +251,19 @@ func _set_time_type(value):
 	time_type = value
 	emit_changed()
 	
-func _set_custom_time(value):
-	custom_time = value
-	emit_changed()
+func _set_custom_time(value: String) -> void:
+	var regex := "^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9],(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$"
+	# using a Regex to define a valid format for the timestamp
+	if validate_time_format(value, regex):
+		custom_time = value
+		emit_changed()
+
+func validate_time_format(time_str: String, regex_str: String) -> bool:
```

Comment:
I don't see the need for it being a separate function as it's not reused anywhere but if you're going to go that route the name should begin with _ since it's private